src/precompiles/native.rs

```diff
@@ -65,34 +59,29 @@ impl Precompile for ExitToNear {
     }
 
     #[cfg(not(feature = "contract"))]
-    fn run(
-        input: &[u8],
-        target_gas: u64,
-        _context: &Context,
-        _is_static: bool,
-    ) -> PrecompileResult {
-        if Self::required_gas(input)? > target_gas {
-            return Err(ExitError::OutOfGas);
+    fn run(input: &[u8], target_gas: Option<u64>, _context: &Context) -> EvmPrecompileResult {
+        if let Some(target_gas) = target_gas {
+            if Self::required_gas(input)? > target_gas {
+                return Err(ExitError::OutOfGas);
+            }
         }
 
-        Ok(PrecompileOutput {
-            output: Vec::new(),
-            cost: 0,
-            logs: Vec::new(),
-            promise: None,
-        })
+        Ok(PrecompileOutput::default().into())
     }
 
     #[cfg(feature = "contract")]
-    fn run(input: &[u8], target_gas: u64, context: &Context, is_static: bool) -> PrecompileResult {
-        if Self::required_gas(input)? > target_gas {
-            return Err(ExitError::OutOfGas);
+    fn run(input: &[u8], target_gas: Option<u64>, context: &Context) -> EvmPrecompileResult {
+        if let Some(target_gas) = target_gas {
+            if Self::required_gas(input)? > target_gas {
+                return Err(ExitError::OutOfGas);
+            }
         }
 
+        // TODO: How to fix this?
```

Comment:
Still need to do something about this. Maybe there is a way from the `context` to determine if this is a static call or not?

Reply:
What if we move the is_static field upstream into context directly. My guess is that it won't break any code for others, and will do the trick for us.

Reply:
Thats the idea at least, to compromise and keep `is_static` upstream. It will break API, but won't really hurt anyone.